internal/controller/reinstall/reinstall_test.go

```diff
@@ -0,0 +1,680 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reinstall
+
+import (
+	"context"
+	"errors"
+
+	"go.uber.org/zap"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/client/interceptor"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/utils/ptr"
+
+	"github.com/stolostron/siteconfig/api/v1alpha1"
+	ci "github.com/stolostron/siteconfig/internal/controller/clusterinstance"
+	"github.com/stolostron/siteconfig/internal/controller/configuration"
+	"github.com/stolostron/siteconfig/internal/controller/deletion"
+	cierrors "github.com/stolostron/siteconfig/internal/controller/errors"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("isRequestProcessed", func() {
+	var (
+		clusterInstance *v1alpha1.ClusterInstance
+	)
+
+	BeforeEach(func() {
+		clusterInstance = &v1alpha1.ClusterInstance{
+			Spec: v1alpha1.ClusterInstanceSpec{
+				Reinstall: &v1alpha1.ReinstallSpec{
+					Generation: "1234",
+				},
+			},
+			Status: v1alpha1.ClusterInstanceStatus{
+				Conditions: []metav1.Condition{},
+				Reinstall:  &v1alpha1.ReinstallStatus{},
+			},
+		}
+	})
+
+	It("should return false if reinstall status is nil", func() {
+		clusterInstance.Status.Reinstall = nil
+		Expect(isRequestProcessed(clusterInstance)).To(BeFalse())
+	})
+
+	It("should return false if no relevant condition exists", func() {
+		Expect(isRequestProcessed(clusterInstance)).To(BeFalse())
+	})
+
+	It("should return false if observed generation does not match reinstall generation", func() {
+		clusterInstance.Status.Reinstall = &v1alpha1.ReinstallStatus{
+			Conditions: []metav1.Condition{
+				*reinstallRequestProcessedConditionStatus(metav1.ConditionTrue, v1alpha1.Completed, "Processed"),
+			},
+			ObservedGeneration: "4567",
+		}
+		Expect(isRequestProcessed(clusterInstance)).To(BeFalse())
+	})
+
+	It("should return true if observed generation matches reinstall generation", func() {
+		clusterInstance.Status.Reinstall = &v1alpha1.ReinstallStatus{
+			Conditions: []metav1.Condition{
+				*reinstallRequestProcessedConditionStatus(metav1.ConditionTrue, v1alpha1.Completed, "Processed"),
+			},
+			ObservedGeneration: "1234",
+		}
+		Expect(isRequestProcessed(clusterInstance)).To(BeTrue())
+	})
+})
+
+var _ = Describe("isNewRequest", func() {
```

Comment:
These tests make it seem like this is just `!isRequestProcessed`. Can you add a test to differentiate the two?

Reply:
I have added tests - but the key difference is to note that `isRequestProcessed ` focusses on `observedGeneration` whereas `isNewRequest` focusses on `inProgressGeneration`.